cpp/test/CMakeLists.txt

```diff
@@ -53,6 +53,7 @@ if(BUILD_CUML_TESTS)
     sg/genetic/node_test.cpp
     sg/genetic/param_test.cu
     sg/handle_test.cu
+    sg/hdbscan_test.cu
```

Comment:
Things didn’t change much in this file, but adding comment to be sure that the test file is still added when merging 21.06 into PR